R/connect.R

```diff
@@ -787,6 +787,36 @@ Connect <- R6::R6Class(
       return(res[["schedules"]])
     },
 
+    # oauth integrations --------------------------------------------
+
+    #' @description Perform an OAuth credential exchange to obtain a
+    #' viewer's OAuth access token.
+    #' @param user_session_token The content viewer's session token. This token
+    #' can only be obtained when the content is running on a Connect server. The token
+    #' identifies the user who is viewing the content interactively on the Connect server.
+    #'
+    #' Read this value from the HTTP header: `Posit-Connect-User-Session-Token`
+    #'
+    #' For example, to read the token from a Shiny session:
+    #'     `user_session_token <- session$request$HTTP_POSIT_CONNECT_USER_SESSION_TOKEN`
+    #' or, to read the token from an inbound HTTP request with Plumbler:
+    #'     `user_session_token <- req$HTTP_POSIT_CONNECT_USER_SESSION_TOKEN`
```

Comment:
- If this is a user-facing function, it needs `#' @export` at the bottom of its `roxygen2` block.
- Consider adding `#' @return Description of the value returned`
- If you haven't already, good practice to check the docs after running `document()` to make sure everything looks right. With the package loaded:

```R
> ?oath_credentials
```